packages/product-components/src/components/SelectAssetModal/SelectAssetModal.tsx

```diff
@@ -55,6 +64,41 @@ export const SelectAssetModal = ({
         $elevation: 0,
     });
 
+    const onScrollEnd = useCallback(() => {
+        setEnd(end + 1000);
+    }, [end]);
+
+    const renderItem = useCallback(
+        (option: AssetProps) => {
+            const {
+                cryptoName,
+                ticker,
+                coingeckoId,
+                symbol,
+                badge,
+                contractAddress,
+                shouldTryToFetch,
+            } = option;
+
+            return (
+                <AssetItem
+                    data-testid={`${dataTestId}/option/${cryptoName}-${symbol}`}
+                    key={`${symbol}${contractAddress ? `-${contractAddress}` : ''}`}
+                    cryptoName={cryptoName}
+                    ticker={ticker}
+                    coingeckoId={coingeckoId}
+                    contractAddress={contractAddress || null}
+                    symbol={symbol}
+                    badge={badge}
+                    shouldTryToFetch={shouldTryToFetch}
+                    handleClick={onSelectAsset}
+                    balance={renderOptionBalance(option)}
+                />
+            );
+        },
+        [dataTestId, onSelectAsset, renderOptionBalance],
```

Comment:
Also, there's was created new arrow function on each render, therefore the `memo` beneath couldn't ever properly work. The `useCallback` here and at the `onScrollEnd` fixed that.